tests/datalab/test_datalab.py

```diff
@@ -981,53 +981,144 @@ def test_pred_probs_precedence(self, pred_probs, random_embeddings):
+    def test_regression_with_predictions(self, lab, regression_data):
+        """Test that the regression issue checks find 9 label issues, based on the
+        predictions of a model.
+
+        Instead of running a model, we use the ground-truth to emulate a perfect model's predictions.
+
+        Testing the default behavior, we expect to find some label issues with a given mean score.
+        Increasing a threshold for flagging issues will flag more issues, but won't change the score.
+        """
+
+        # Use ground-truth to emulate a perfect model's predictions
+        y_pred = regression_data["true_y"]
+
+        lab.find_issues(pred_probs=y_pred)
+        summary = lab.get_issue_summary()
+
+        issues = lab.get_issues("label")
+        issue_ids = issues.query("is_label_issue").index
+        expected_issue_ids = regression_data["error_idx"]
+
+        # jaccard similarity
+        intersection = len(list(set(issue_ids).intersection(set(expected_issue_ids))))
+        union = len(set(issue_ids)) + len(set(expected_issue_ids)) - intersection
+        assert float(intersection) / union > 0.8
+
+        # FPR
+        fpr = len(list(set(issue_ids).difference(set(expected_issue_ids)))) / len(issue_ids)
+        assert fpr == 0.0
+
+        # Try running with a different threshold
+        lab.find_issues(pred_probs=y_pred, issue_types={"label": {"threshold": 0.4}})
+        issues = lab.get_issues("label")
+        issue_ids = issues.query("is_label_issue").index
+
+        intersection = len(list(set(issue_ids).intersection(set(expected_issue_ids))))
+        union = len(set(issue_ids)) + len(set(expected_issue_ids)) - intersection
+        assert float(intersection) / union > 0.3
+
+    def test_regression_with_model_and_features(self, lab, regression_data):
+        """Test that the regression issue checks find label issue with another model."""
+        from sklearn.linear_model import RANSACRegressor
+
+        model = RANSACRegressor(random_state=SEED)
+        X = regression_data["X"]
+        issue_types = {"label": {"clean_learning_kwargs": {"model": model, "seed": SEED}}}
+        lab.find_issues(features=X, issue_types=issue_types)
+
+        issues = lab.get_issues("label")
+        issue_ids = issues.query("is_label_issue").index
+        expected_issue_ids = regression_data[
+            "error_idx"
+        ]  # Set to 5% of the data, but random noise may be too small to detect
+
+        # jaccard similarity
+        intersection = len(list(set(issue_ids).intersection(set(expected_issue_ids))))
+        union = len(set(issue_ids)) + len(set(expected_issue_ids)) - intersection
+        assert float(intersection) / union > 0.3
 
-    assert "label" in summary["issue_type"].values
-    assert (summary[summary["issue_type"] == "label"]["num_issues"] == 40).all()
+        # FPR
+        fpr = len(list(set(issue_ids).difference(set(expected_issue_ids)))) / len(issue_ids)
+        assert fpr < 0.3
```

Comment:
open GH issue to improve this `fpr` in the future

Reply:
- [ ] Improve the stability of results from end-to-end tests of Datalab with label error-detection for regression tasks